src/gamma.jl

```diff
@@ -50,6 +47,49 @@ function _gamma(x::Float64)
     return z * p / q
 end
 
+function gamma(_x::Float32)
+    x = Float64(_x)
+    if _x < 0
+        s = sinpi(x)
+        s == 0 && throw(DomainError(_x, "NaN result for non-NaN input."))
+        x = 1 - x
+    end
+    if x < 5
+        z = 1.
+        while x>1
+            x -= 1
+            z *= x
+        end
+        num = evalpoly(x, (1.0, 0.41702538904450015, 0.24081703455575904, 0.04071509011391178, 0.015839573267537377))
+        den = x*evalpoly(x, (1.0, 0.9942411061082665, -0.17434932941689474, -0.13577921102050783, 0.03028452206514555))
+        res = z*num/den
+    else
+        x -= 1
+        w = evalpoly(inv(x), (2.506628299028453, 0.20888413086840676, 0.008736513049552962, -0.007022997182153692, 0.0006787969600290756))
+        res = @fastmath sqrt(x) * exp(log(x*1/ℯ) * x) * w
+    end
+    return Float32(_x<0 ? π / (s * res) : res)
+end
+
+function gamma(_x::Float16)
+    x = Float32(_x)
+    if _x < 0
+        s = sinpi(x)
+        s == 0 && throw(DomainError(_x, "NaN result for non-NaN input."))
+        x = 1 - x
+    end
+    x>13 && return Float16(ifelse(_x>0, Inf32, 0f0))
+	z = 1f0
+	while x>1
+		x -= 1
+		z *= x
+	end
+	num = evalpoly(x, (1.0f0, 0.4170254f0, 0.24081704f0, 0.04071509f0, 0.015839573f0))
+	den = x*evalpoly(x, (1.0f0, 0.9942411f0, -0.17434932f0, -0.13577922f0, 0.030284522f0))
+    return Float16(_x<0 ? Float32(π)*den/(s*z*num) : z*num/den)
+end
+
+_gamma(x) = gamma(x) #easier than fixing this in other places.
```

Comment:
I think https://github.com/JuliaMath/Bessels.jl/blob/fb17d4e0a11784145460e3dc46251b4d3a3885f2/src/gamma.jl#L7 is the only instance of `_gamma(x)` so might be best to just change that. I don't think anything in the bessel routines uses `_gamma`